vtex/toolbelt
# Feature/conflicts in promote

#### What is the purpose of this pull request?
Add conflict handling to `vtex workspace promote`. Also, some log/success messages were updated to fit the designer's prototype, you can check it out [here](https://www.figma.com/file/SR9a9fT1W6dh2EUri6RPGL/Commands-Flow?node-id=12%3A106) and about the colors [here](https://www.figma.com/file/GpNy0htTpJ7uvw7WKqGdLH/Toolbelt-DX-review?node-id=93%3A21).

#### What problem is this solving?
Changes in workspace are being lost when `vtex workspace promote` is run if conflicts are detected, as conflicts are fixed only when the workspace is opened in the browser. Therefore, if the user run promote without previously opening the workspace in browser their changes are replaced with the masters (_master wins_ strategy), but we want to use the _mine wins_ strategy -> this is the conflict handling implemented in this PR.

You can see more on this [task](https://vtex-dev.atlassian.net/browse/DT-18?atlOrigin=eyJpIjoiNGViMTE4NDg4ZjhlNDU5OWJlN2ZmY2JmNTExOTgyZTYiLCJwIjoiaiJ9).

#### How should this be manually tested?
Run the project in this branch, then after running `yarn watch` follow the steps:

1. Create a new production workspace: `vtex-test workspace create your-workspace-here --production`
2. Use this new workspace: `vtex-test workspace use your-workspace-here`
3. Make some change using your workspace https://workspace--storecomponents.myvtex.com/admin/cms/site-editor
4. Make some change in the same component now using master https://storecomponents.myvtex.com/admin/cms/site-editor (Remember not to open your workspace in the browser)
5. (Optional) If you want to test the request to see if conflict is detected: `http://infra.io.vtex.com/vbase/v0/storecomponents/your-workspace-here/buckets/vtex.pages-graphql/userData/files/store/content.json` using this header: `X-Vtex-Detect-Conflicts: true`. It should return 409.
6. Test the `vtex-test workspace promote` (run twice to answer no and then yes to the confirmation prompt to see both flows)
7. Check https://storecomponents.myvtex.com/ to see if the changes you make on your workspace are now on master.

#### Screenshots or example usage
- "No" flow:
<img width="961" alt="Screen Shot 2020-08-04 at 13 57 52" src="https://user-images.githubusercontent.com/19983991/89322374-82517a80-d65a-11ea-861e-4b12eaa29bf3.png">

- "Yes" flow:
<img width="983" alt="Screen Shot 2020-08-04 at 13 58 34" src="https://user-images.githubusercontent.com/19983991/89322440-985f3b00-d65a-11ea-8a97-7e48c5bf0de7.png">


#### Types of changes

#### Chores checklist
